src/components/Navigation/TopBar/Help/FeedbackForm/FeedbackForm.tsx

```diff
@@ -31,50 +44,96 @@ const FeedbackForm: FC<FeedbackFormProps> = ({ onClose, selectedType }) => {
-    );
+  const {
+    mutateAsync: slackFileUpload,
+    isLoading: isFileUploadLoading,
+    isSuccess: isFileUploadSuccess,
+  } = useMutation(['slackFileUpload', feedbackContent], (formData: FormData) =>
+    PortalService.fileUpload(formData)
+  );
 
-  const { mutateAsync: slackPostMessage, isLoading: isPostMessageLoading } =
-    useMutation(['slackPostMessage', feedbackContent], (formData: FormData) =>
-      PortalService.postmessage(formData)
-    );
+  const {
+    mutateAsync: slackPostMessage,
+    isLoading: isPostMessageLoading,
+    isSuccess: isPostMessageSuccess,
+  } = useMutation(['slackPostMessage', feedbackContent], (formData: FormData) =>
+    PortalService.postmessage(formData)
+  );
 
-  const { mutateAsync: serviceNowIncident, isLoading: isServiceNowLoading } =
-    useMutation(
-      ['serviceNowIncident', feedbackContent],
-      async (serviceNowDto: ServiceNowIncidentRequestDto) =>
-        PortalService.createIncident(serviceNowDto)
-    );
+  const {
+    mutateAsync: serviceNowIncident,
+    isLoading: isServiceNowLoading,
+    isSuccess: isServiceNowSuccess,
+    data: response,
+  } = useMutation(
+    ['serviceNowIncident', feedbackContent],
+    async (formData: FormData) => PortalService.createIncident(formData)
+  );
+
+  const relevantRequestsIsSuccess = useMemo(() => {
+    const booleanArray: boolean[] = [];
+    if (
+      feedbackContent?.attachments &&
+      feedbackContent?.attachments?.length > 0
+    ) {
+      booleanArray.push(isFileUploadSuccess);
+    }
+    if (selectedType === FeedbackEnum.BUG) {
+      booleanArray.push(isServiceNowSuccess);
+    }
+    booleanArray.push(isPostMessageSuccess);
+    return !booleanArray.includes(false) && booleanArray.length > 0;
+  }, [
+    feedbackContent?.attachments,
+    isFileUploadSuccess,
+    isPostMessageSuccess,
+    isServiceNowSuccess,
+    selectedType,
+  ]);
 
   const requestIsLoading = useMemo(() => {
     return isPostMessageLoading || isFileUploadLoading || isServiceNowLoading;
   }, [isFileUploadLoading, isPostMessageLoading, isServiceNowLoading]);
 
+  const serviceNowNumber = useMemo(() => {
+    if (selectedType === FeedbackEnum.BUG) {
+      if (!response?.number) return 'Not found';
+      return response.number;
+    }
+  }, [response?.number, selectedType]);
+
   const updateFeedback = (
     key: keyof FeedbackContentType,
-    newValue: string | SeverityOption | FileWithPath[] | boolean
+    newValue: string | UrgencyOption | FileWithPath[] | boolean
   ) => {
-    setFeedbackContent({ ...feedbackContent, [key]: newValue });
+    if (key === 'attachments' && feedbackContent.attachments) {
+      setFeedbackContent((prev) => {
+        return {
+          ...prev,
+          [key]: [...(newValue as FileWithPath[])],
+        };
+      });
+    } else {
+      setFeedbackContent({ ...feedbackContent, [key]: newValue });
+    }
+  };
+
+  const getUrgencyNumber = (urgency: UrgencyOption) => {
```

Comment:
This is just a util, would prefer to have it outside the function in a separate file